pkg/services/scm_provider/utils.go

```diff
@@ -52,6 +71,23 @@ func matchFilter(ctx context.Context, provider SCMProviderService, repo *Reposit
 		return false, nil
 	}
 
+	if filter.PullRequestBranchMatch != nil && !filter.PullRequestBranchMatch.MatchString(repo.Branch) {
+		return false, nil
+	}
+
+	if filter.PullRequestLabelMatch != nil {
```

Comment:
With adding the `PullRequest` field to the `Repository` struct, we can keep this and range through `repo.PullRequest.Labels` instead of `repo.Labels`.
https://github.com/argoproj/applicationset/pull/469#discussion_r819411302

Reply:
Yep, I think that's the way to go.

Will need to consider how to treat a pull-request filter placed on an SCM generator where pull request fetching isn't enabled.